### ISSUE TITLE
Parse inv messages.

### ISSUE THREAD

**Issue by hdevalence** [ZcashFoundation/zebra#24]
We should be able to parse inventory messages.

**Comment by hdevalence** [comment]
I think this is blocked on #21, because in order to represent the ["inventory items"](https://en.bitcoin.it/wiki/Protocol_documentation#Inventory_Vectors) as
```
/// An inventory item which refers to some advertised or requested data.
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
enum InventoryItem {
    Error,
    Tx(TxHash),
    Block(BlockHash),
    FilteredBlock(BlockHash),
}
```
we need the `TxHash`, `BlockHash` types to exist already.

**Comment by hdevalence** [comment]
@dconnolly can I move the existing `InventoryType`/`InventoryVector` structs out of `zebra-chain` and into `zebra-network`? I *think* that they're just used for the network protocol, so it seems like a better home.

**Comment by dconnolly** [comment]
> @dconnolly can I move the existing `InventoryType`/`InventoryVector` structs out of `zebra-chain` and into `zebra-network`? I _think_ that they're just used for the network protocol, so it seems like a better home.

Yes I think this is correct, `Inventory*` look like message or general metadata rather than an independent type. 👍

**Comment by hdevalence** [comment]
Closed by the combination of #31 and #38 (which replaced some stub types).
